lib/modules/dosomething/dosomething_signup/includes/SignupsController.php

```diff
@@ -1,73 +1,59 @@
 <?php
 
-/**
- * @file
- * Provides classes for the Signup Entity.
- */
+class SignupsController extends EntityAPIController {
```

Comment:
Definitely not for this PR, but I wonder if it'd improve testability to have the controller classes completely independent of Drupal, and then just have a `EntityAPIRouter` class or something that acts as the "glue" to connect it. Then we could potentially run PHPUnit tests without bootstrapping the whole monstrosity.